vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -47,16 +49,66 @@ type InstallerController struct {
 
 	// installerPodImageFn returns the image name for the installer pod
 	installerPodImageFn func() string
+
+	// installStateFunc derives the static pod install state of a node
+	staticPodStateFunc StaticPodStateFunc
+}
+
+// StaticPodState is the status of a static pod that has been installed to a node.
+type StaticPodState int
+
+const (
+	// staticPodStatePending means that the installed static pod is not up yet.
+	StaticPodStatePending = StaticPodState(iota)
+	// StaticPodStateReady means that the installed static pod is ready.
+	StaticPodStateReady
+	// StaticPodStateFailed means that the static pod installation of a node has failed.
+	StaticPodStateFailed
+)
+
+const deploymentIDLabel = "deployment-id"
+
+// StaticPodStateFunc derives the status of the installed static pod. It returns the state, an array of error strings
+// in case of StaticPodStateFailed and an error in case the state couldn't be checked.
+type StaticPodStateFunc func(deploymentID int32, nodeName string) (StaticPodState, []string, error)
```

Comment:
seems like deploymentID can be an output,not an input.  There is only one possible per node.

Reply:
We have to know which one is the target. There can be an old static pod. Hence, deployment id must be an input. Will rename it.

Reply:
`targetDeploymentID`